packages/story-editor/src/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@web-stories-wp/i18n';
+import { Input } from '@web-stories-wp/design-system';
+import { useState, useCallback } from 'react';
+import styled from 'styled-components';
+import {
+  createResource,
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+  getImageDimensions,
+  getVideoDimensions,
+  hasVideoGotAudio,
+  getVideoLength,
+} from '@web-stories-wp/media';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../../app';
+import Dialog from '../../../../../dialog';
+import useLibrary from '../../../../useLibrary';
+import {
+  useUploadVideoFrame,
+  getPosterName,
+} from '../../../../../../app/media/utils';
+import { withProtocol } from '../../../../../../utils/url';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const {
+    allowedFileTypes,
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+
+  const [errorMsg, setErrorMsg] = useState(false);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
+    description
+  );
+  const [link, setLink] = useState('');
+
+  const getFileInfo = useCallback(
+    (value = link) => {
```

Comment:
First check for `isValidURL`?

Reply:
Hm, link not being valid would come out when clicking "Insert", too. Perhaps that's enough?

If we're checking if the link if invalid while typing then there should be a different error message for that, maybe just "Please ensure the link is valid"?

Thoughts?

Reply:
Oh right, I meant adding an `isValidURL` check in `onInsert()` callback below.

I don't see any client-side validation of the image url using `isValidUrl` here anywhere. Only validation being done is `value.split(/[#?]/)[0].split('.').pop().trim()`, which doesn't work for all cases anyway as discussed.

So I'd expect an actual check whether the URL is valid or not.

Reply:
Some unit tests might help here.

A URL like `http://.jpeg` would not cause an error with this regex, but the `isValidUrl` should catch it.

LMK if I need to rephrase this :)

Reply:
Yep, understood, it's just that getting the media dimensions will fail in case of invalid URL and then it'll show the error message anyway. However, we could also separately check first, too, before starting getting the dimensions at all.

Reply:
Added the check, however, in testing, `isValidUrl` did not catch the specific case of `https://.jpeg` and it was still caught when getting the dimensions instead 🤔

Reply:
Hmm that function might be a bit lax then. In any case, such client-side validation is needed. Later we'll add server-side validation too.